runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -154,69 +156,116 @@ end
 --- This will run the injection query for this language to
 --- determine if any child languages should be created.
 ---
----@return TSTree[]
+---@param opts table|nil Options table:
+---       - timeout (integer|nil): When provided parsing will run asynchronously
+---         by segmenting the parse over Nvim's event loop where each segment
+---         will run for at most this value in ms.
```

Comment:
This extra info would be valuable in this doc:

> A parse is never cancelled because of this, but it will mean TSParser:parse becomes non-blocking and won't return a result.